### PR TITLE
[GAL-3029] Mobile Notification Upsell

### PR DESCRIPTION
**Pull request by tbezman**
image

Used AsyncStorage to ensure the user only ever sees this once.

### CODE REVIEW

File: apps/mobile/index.js [hunk 1]
```diff
@@ -42,6 +43,7 @@ LogBox.ignoreLogs([
 polyfillEncoding();
 polyfillFetch();
 polyfillReadableStream();
+polyfillUrl();
```
Comment (tbezman) [inline apps/mobile/index.js:46]: we need this polyfill for the notification package

File: apps/mobile/src/components/DevMenuItems.tsx [hunk 2]
```diff
@@ -8,6 +9,12 @@ export function DevMenuItems() {
 
   useEffect(() => {
     registerDevMenuItems([
+      {
+        name: 'Clear Async Storage',
+        callback: () => {
+          AsyncStorage.clear();
+        },
+      },
```
Comment (tbezman) [inline apps/mobile/src/components/DevMenuItems.tsx:17]: Helpful for debugging

File: apps/mobile/src/components/Notification/NotificationList.tsx [hunk 3]
```diff
@@ -80,22 +80,24 @@ export function NotificationList({ queryRef }: Props) {
...
+            ),
+            ConnectionHandler.getConnectionID(
+              query.viewer?.id,
+              'NotificationsFragment_notifications'
+            ),
+          ]);
+        }
+      };
+    }, [clearNotification, query.viewer?.id, query.viewer?.user?.dbid])
+  );
```
Comment (tbezman) [inline apps/mobile/src/components/Notification/NotificationList.tsx:100]: This was firing way too many times w/o the use callback

File: apps/mobile/src/components/Notification/NotificationRegistrar.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  const relayEnvironment = useRelayEnvironment();
+
+  useEffect(() => {
+    registerNotificationToken({ shouldPrompt: false, relayEnvironment });
+  }, [relayEnvironment]);
+
+  const navigation = useNavigation<RootStackNavigatorProp>();
+  useEffect(() => {
+    const cleanup = Notifications.addNotificationResponseReceivedListener(() => {
+      navigation.navigate('MainTabs', {
```
Comment (tbezman) [inline apps/mobile/src/components/Notification/NotificationRegistrar.tsx:19]: navigate to the notifications screen when the user presses a notification. 

use a random fetch key so we definitely get a refresh

**Review by kaitoo1 — APPROVED** [review]

**Review by Robinnnnn — APPROVED** [review]

**Pull request merged**